File: src/com/colacelli/irclib/connection/listeners/OnPrivateNoticeMessageListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.connection.Connection;
import com.colacelli.irclib.messages.PrivateNoticeMessage;

public interface OnPrivateNoticeMessageListener extends Listener {
    Type TYPE = Type.PRIVATE_NOTICE;

    void onNoticeMessage(Connection connection, PrivateNoticeMessage message);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/main/kotlin/com/colacelli/irclib/connection/listeners/OnPrivateNoticeMessageListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.connection.Connection
import com.colacelli.irclib.messages.PrivateNoticeMessage

interface OnPrivateNoticeMessageListener : Listener {
    fun onNoticeMessage(connection: Connection, message: PrivateNoticeMessage)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.PRIVATE_NOTICE
    }
}
